File: src/main/java/com/hyecheon/antteshop/entity/BaseEntity.java
package com.hyecheon.antteshop.entity;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;
import org.hibernate.annotations.Where;
import org.springframework.data.annotation.CreatedDate;
import org.springframework.data.annotation.LastModifiedDate;
import org.springframework.data.jpa.domain.support.AuditingEntityListener;

import javax.persistence.Column;
import javax.persistence.EntityListeners;
import javax.persistence.MappedSuperclass;
import java.time.LocalDateTime;

/**
 * User: hyecheon lee
 * Email: rainbow880616@gmail.com
 * Date: 2021/08/14
 */

@Data
@MappedSuperclass
@EntityListeners({AuditingEntityListener.class})
@Where(clause = "deleted = 0")
public abstract class BaseEntity {

    @CreatedDate
    @Column(nullable = false, updatable = false)
    private LocalDateTime createdAt = LocalDateTime.now();

    @LastModifiedDate
    @Column(nullable = false)
    private LocalDateTime updatedAt = LocalDateTime.now();

    @Column(name = "isDeleted", nullable = false)
    private boolean deleted;

}

File: src/main/kotlin/com/hyecheon/antteshop/domains/entity/BaseEntity.kt
package com.hyecheon.antteshop.domains.entity

import lombok.Data
import org.hibernate.annotations.Where
import org.springframework.data.annotation.CreatedDate
import org.springframework.data.annotation.LastModifiedDate
import org.springframework.data.jpa.domain.support.AuditingEntityListener
import java.time.LocalDateTime
import javax.persistence.Column
import javax.persistence.EntityListeners
import javax.persistence.MappedSuperclass

/**
 * User: hyecheon lee
 * Email: rainbow880616@gmail.com
 * Date: 2021/08/14
 */
@MappedSuperclass
@EntityListeners(AuditingEntityListener::class)
@Where(clause = "deleted = 0")
abstract class BaseEntity {
    @CreatedDate
    @Column(nullable = false, updatable = false)
    var createdAt = LocalDateTime.now()

    @LastModifiedDate
    @Column(nullable = false)
    var updatedAt = LocalDateTime.now()

    @Column(name = "isDeleted", nullable = false)
    var deleted: Boolean = false
}